### ISSUE TITLE
tab colors not working as intended 

### ISSUE THREAD

**Issue by Strynx79** [sagars007/starry-fox#14]
![image](https://github.com/user-attachments/assets/64b50b26-0c16-4fa8-8b87-fd5f7dcf3749)
i edited the code to show the find box but i cant find a fix for this issue

**Comment by Strynx79** [comment]
![image](https://github.com/user-attachments/assets/5151d6be-cfcb-4fb1-acfa-7c28d1ed7f5a)
 how it is supposed to look as shown by author

**Comment by Vasilo58** [comment]
Same Problem here

**Comment by luanaraujo** [comment]
Same Problem here

**Comment by A0D1I2L3** [comment]
> ![image](https://github.com/user-attachments/assets/64b50b26-0c16-4fa8-8b87-fd5f7dcf3749) i edited the code to show the find box but i cant find a fix for this issue

Try these in userChrome.css

```
/* === 1. Close button hidden by default, appears on hover === */
.tabbrowser-tab:not([pinned]) .tab-close-button {
  display: -moz-box !important;
  opacity: 0 !important;
  visibility: hidden !important;
  transition: opacity 200ms ease-in-out !important;
}
.tabbrowser-tab:not([pinned]):hover .tab-close-button {
  opacity: 1 !important;
  visibility: visible !important;
}

/* === 2. Fix tab label mask size === */
.tabbrowser-tab {
  --tab-label-mask-size: 1em !important;
}

/* === 3. Smooth transition for close icon fill === */
.close-icon {
  transition: fill 300ms cubic-bezier(0.645, 0.045, 0.355, 1) !important;
}
.close-icon:hover {
  fill: #FF0000 !important;
}

/* === 4. Selected tab gradient background === */
.tabbrowser-tab[selected="true"] .tab-background {
  background: linear-gradient(to right,
    rgba(159, 26, 255, 0.3),
    rgba(138, 70, 255, 0.3),
    rgba(118, 93, 255, 0.3),
    rgba(93, 125, 255, 0.3),
    rgba(62, 140, 255, 0.3),
    rgba(28, 153, 255, 0.3),
    rgba(0, 165, 255, 0.3),
    rgba(0, 180, 255, 0.3),
    rgba(0, 195, 255, 0.3),
    rgba(0, 208, 255, 0.3),
    rgba(0, 221, 255, 0.3)
  ) !important;
  border-radius: 4px !important;
  border: none !important;
}

/* === 5. Container tab side indicators and background with thickness and background change === */
#tabbrowser-tabs .tab-context-line {
  display: none !important;
}

.tabbrowser-tab[class*="identity-color-"] {
  border-left: 2px solid var(--identity-icon-color, #7C4DFF) !important;
  border-right: 2px solid var(--identity-icon-color, #7C4DFF) !important;
  border-radius: 4px !important;
  background-clip: padding-box !important;
  background-color: rgba(0, 0, 0, 0.05) !important;
  transition: border-width 0.15s ease-in-out, background-color 0.15s ease-in-out !important;
}

.tabbrowser-tab[selected="true"][class*="identity-color-"] {
  border-left-width: 4px !important;
  border-right-width: 4px !important;
  background-color: rgba(0, 0, 0, 0.1) !important;
}

/* === 6. Container tab label color constant regardless of selection === */
.tabbrowser-tab[class*="identity-color-"] .tab-label {
  color: var(--identity-tab-color, #7C4DFF) !important;
  text-shadow: none !important;
}

/* Fallback text-shadow trick */
.tabbrowser-tab[class*="identity-color-"] .tab-label {
  color: transparent !important;
  text-shadow: 0 0 0 var(--identity-tab-color, #7C4DFF) !important;
}

/* === 7. Unread and busy tabs (via extensions like Tab Flag) === */
.tabbrowser-tab[titlechanged]:not([selected="true"]) .tab-label,
.tabbrowser-tab[busy]:not([selected="true"]) .tab-label {
  font-style: italic !important;
  color: #00DDFF !important;
}

```


Remember that those colours are container tab colours , so open the tabs in new container to test them

**Comment by A0D1I2L3** [comment]
![Image](https://github.com/user-attachments/assets/a844fde2-26c8-4aa9-8dae-5cfb65a5af37)

Here is a comparison image

**Comment by pompersuf** [comment]
and do i replace stuff in the userchrome.css or add to it because i dont know a whole lot about css stuff would appreciate the extra help and which ones do i need to replace for this to start working?
Update- literally 10 mins later using common sense and my deductive skills i got the tab colors to work again but all i need now is the info for the search bar gradient to work as well is there any way i can get that? you have been a massive help thank you

**Comment by A0D1I2L3** [comment]
Glad to know that tab colors worked
I am attaching my userChrome.css as a txt file so you can check the style code for Search bar 

[userChrome.txt](https://github.com/user-attachments/files/20739140/userChrome.txt)

look for classes containing the word urlbar

When you open a new firefox window searchbar will have a gradient background , after first search the search bar will have a gradient border on the bottom

**Comment by pompersuf** [comment]
your so goated my dude big thanks G
quick update- im looking through both and they seem to be exactly the same for the urlbar classes weirdly enough and i cant seem to find anything in regards to the urlbar that is different in mine that i would have to change to get it to work

**Comment by A0D1I2L3** [comment]
in that case maybe replace your file with mine and check

**Comment by pompersuf** [comment]
i  literally inserted yours and nothing has changed and now im sad maybe its not meant to be 😭
Edit- i am now noticing that i have the toolbar back but it does not have anything on it and its purple for some reason

**Comment by A0D1I2L3** [comment]
Yeah the toolbar was also fixed in my styles , its empty cuz you have added no bookmarks

check that When you open a new firefox window searchbar will have a gradient background , after first search the search bar will have a gradient border on the bottom

it will not have a gradient background forever 
you  need to change more styles for that

**Comment by A0D1I2L3** [comment]
Dont get sad :) , The styles i gave did the following

- When you open a new firefox **window** searchbar will have a gradient background , 
- after first search the search bar will have a gradient border on the bottom 
-  the searchbar wouldnt have a gradient forever

I am attaching new styles to match what you need 

[userChrome.txt](https://github.com/user-attachments/files/20760550/userChrome.txt)

![Image](https://github.com/user-attachments/assets/7fc05bd8-d77c-4f4e-bcd5-dddf6a56a9ba)
 i made toolbar purple so i can easily spot and right click it  , you can change it if you want :)

**Comment by pompersuf** [comment]
ok i litereally just copy and pasted the whole thing and it got the search bar to work but now the toolbar is there but it is not showing my bookmarks or anything on it do you have a fix for that and everything will be finally complete and thank you man hor the major help brother 🙏😍😘

**Comment by A0D1I2L3** [comment]
go to bookmarks manager and check if your bookmarks are in bookmark toolbar , if they are in bookmarks menu drag them to bookmarks toolbar else add them to bookmarks toolbar

**Comment by pompersuf** [comment]
yeah they are there or atleast i think  it is 

![Image](https://github.com/user-attachments/assets/c7ce7204-54c3-4b93-b277-dd8904369d13)
-update bruh i fixed it by accident and now i feel fucking stupid lmfao i just dragged and dropped the bookmarks into the toolbar and bam it worked
and thanks for being patient and helping man i really do appreciate it man
